core/store/src/trie/trie_storage.rs

```diff
@@ -143,3 +144,24 @@ impl TrieStorage for TrieCachingStorage {
         Some(self)
     }
 }
+
+/// Runtime counts the number of touched trie nodes for the purpose of gas calculation.
+/// Trie increments it on every call to TrieStorage::retrieve_raw_bytes()
+#[derive(Default)]
+pub struct TouchedNodesCounter {
```

Comment:
Our trie code is not multithreaded so using atomics is not necessary.

Reply:
it's not but Trie class has to be Send+Sync, will have to refactor at some point